vulnerabilities/importers/elixir_security.py

```diff
@@ -100,40 +92,44 @@ def get_versions_from_range(self, version_list, pkg_name):
         return pkg_versions
 
     def process_file(self, path):
-        with open(path) as f:
-            yaml_file = yaml.safe_load(f)
-            pkg_name = yaml_file["package"]
-            safe_pkg_versions = []
-            if yaml_file.get("unaffected_versions"):
-                safe_pkg_versions = self.get_versions_from_range(
-                    yaml_file["patched_versions"] +
-                    yaml_file["unaffected_versions"],
-                    pkg_name,
-                )
-            else:
-                safe_pkg_versions = self.get_versions_from_range(
-                    yaml_file["patched_versions"], pkg_name
-                )
-
-            cve_id = "CVE-"+str(yaml_file["cve"])
-            safe_purls = []
-            if safe_pkg_versions is not None:
-                safe_purls = {
-                    PackageURL(name=pkg_name, type="hex", version=version)
-                    for version in safe_pkg_versions
-                }
-
-            vuln_reference = [
-                Reference(
-                    reference_id=yaml_file["id"],
-                    url=yaml_file["link"],
-                )
-            ]
-
-            return Advisory(
-                summary=yaml_file["description"],
-                impacted_package_urls=[],
-                resolved_package_urls=safe_purls,
-                cve_id=cve_id,
-                vuln_references=vuln_reference,
+        yaml_file = load_yaml(path)
+        pkg_name = yaml_file["package"]
+        safe_pkg_versions = []
+        if yaml_file.get("unaffected_versions"):
+            safe_pkg_versions = self.get_versions_from_range(
+                yaml_file["patched_versions"] +
+                yaml_file["unaffected_versions"],
+                pkg_name,
+            )
+        else:
```

Comment:
It would be better to be defensive here, (since these things change all the time) . How about eliminating this `if-else` clause and use 

```
    safe_pkg_versions = self.get_versions_from_range(
                yaml_file.get("patched_versions", []) +
                yaml_file.get("unaffected_versions", []),
                pkg_name,
            )
```
It's more concise and defensive(which is better in this context )

Reply:
Yes , great suggestion

Reply:
@sbs2001 This is not working .I am getting <code>unsupported operation + on NoneType and List</code>error. I tried even type casting list. Should I instead switch to something like 
``` json
if yaml_file.get("patched_versions"):
            if yaml_file.get("unaffected_versions"):
                safe_pkg_versions = self.get_versions_from_range(
                    yaml_file["patched_versions"] + yaml_file["unaffected_versions"],
                    pkg_name,
                )
            else:
                safe_pkg_versions, vuln_pkg_versions = self.get_versions_from_range(
                    yaml_file["patched_versions"],
                    pkg_name,
                )
```

Reply:
@tushar912  this is caused because `yaml_file` has the key `"patched_versions"` , so it's not relying on the `get`'s default value. If you see the failing case, most likely you would find `yaml_file["patched_versions"]: None`  . 

A type cast wouldn't solve this. 

Do some handling of `None` value and this will be fixed